src/main/java/heylo/tasks/Task.java

```diff
@@ -0,0 +1,224 @@
+    }
+
+    /**
+     * Adds the given task to the task list if valid and increments task count.
+     *
+     * @param task Task added by user.
+     */
+    public static void addToList(Task task) {
+        if (task.description == null) {
+            System.out.println(" Please enter the description as well!");
+            System.out.println(" Command format: task-type task-description /by task-duration OR /at task-deadline");
+            return;
+        }
+
+        if (task instanceof Event) {
+            Event event = (Event) task;
+            if (event.duration == null) {
+                System.out.println(" Please enter the duration as well!");
+                System.out.println(" Command format: task-type task-description /by YYYY-MM-DD");
+                return;
+            }
+        }
+
+        if (task instanceof Deadline) {
+            Deadline taskWithDeadline = (Deadline) task;
+            if (taskWithDeadline.deadline == null) {
+                System.out.println(" Please enter the deadline as well!");
+                System.out.println(" Command format: task-type task-description /at YYYY-MM-DD");
+                return;
+            }
+        }
+
+        if (taskCount == 0) {
+            System.out.println(" Added! Now you have 1 item in your tasks list.");
+        } else {
+            System.out.println(" Added! Now you have " + (taskCount + 1) + " items in your tasks list.");
+        }
+        taskList.add(task);
+        System.out.println("  " + taskList.get(taskCount).toString());
+        taskCount++;
+    }
+
+    /**
+     * Removes the task from the task list if it exists.
+     *
+     * @param index int Index of the task in the displayed list of tasks.
+     */
+    public static void removeFromList(int index) {
+        try {
+            String outputTask = "  " + taskList.get(index).toString();
+            taskList.remove(index);
+            System.out.println(" Okay, I've deleted this task.");
+            System.out.println(outputTask);
+            taskCount--;
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println(" This task does not exist.");
+        }
+    }
+
+    /**
+     * Populates the task list with the previously saved tasks.
+     */
+    public static void getSavedTasks() {
+        taskList = FilesReader.getTaskListFromFile();
+        taskCount = taskList.size();
+    }
+
+    /**
+     * Saves the tasks in the task list for later retrieval.
+     */
+    public static void saveTasks() {
+        for (Task task : taskList) {
+            if (task instanceof Todo) {
+                Todo todo = (Todo) task;
+                FilesWriter.writeToFile("T : " + (todo.isDone ? "1 : " : "0 : ") + todo.description + "\n");
+            } else if (task instanceof Event) {
+                Event event = (Event) task;
+                FilesWriter.writeToFile("E : " + (event.isDone ? "1 : " : "0 : ") + event.description + " : " + event.duration + "\n");
```

Comment:
Perhaps a line break here? 

Additionally, it would be better if you can avoid complicated expressions.